test/adapter/json_api/include_param_test.rb

```diff
@@ -0,0 +1,166 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        class IncludeParamTest < ActiveSupport::TestCase
+          IncludeParamAuthor = Class.new(::Model)
+
+          class CustomCommentLoader
+            def all
+              [{ foo: 'bar' }]
+            end
+          end
+
+          class TagSerializer < ActiveModel::Serializer
+            attributes :id, :name
+          end
+
+          class IncludeParamAuthorSerializer < ActiveModel::Serializer
+            class_attribute :comment_loader
+
+            has_many :tags, serializer: TagSerializer do
+              link :self, '//example.com/link_author/relationships/tags'
+              include_data :via_include_parameter
+            end
+
+            has_many :unlinked_tags, serializer: TagSerializer do
+              include_data :via_include_parameter
+            end
+
+            has_many :posts, serializer: PostWithTagsSerializer do
+              include_data :via_include_parameter
+            end
+            has_many :locations do
+              include_data :via_include_parameter
+            end
+            has_many :comments do
+              include_data :via_include_parameter
+              IncludeParamAuthorSerializer.comment_loader.all
+            end
+          end
+
+          def setup
+            IncludeParamAuthorSerializer.comment_loader = Class.new(CustomCommentLoader).new
+            @tag = Tag.new(id: 1337, name: 'mytag')
+            @author = IncludeParamAuthor.new(
+              id: 1337,
+              tags: [@tag]
+            )
+          end
+
+          def test_relationship_not_loaded_when_not_included
```

Comment:
I think this test is the central feature in this PR, at least for the JSONAPI adapter